### PR TITLE
Data seeding

### PR DESCRIPTION
**Pull request by mflinn92**
Hi all, if any one is up for a code review feel free to take a look

### CODE REVIEW

File: db/index.js [hunk 1]
```diff
@@ -1,5 +1,28 @@
 const mongoose = require('mongoose');
 
 mongoose.connect('mongodb://localhost/BTetsy' ,{useNewUrlParser: true, useUnifiedTopology: true});
-const db = mongoose.connection;
-db.once('open', () => console.log('mongoose is connected'));
+
+const productDetailsSchema = new mongoose.Schema({
```
Comment (matt-leyrat) [inline db/index.js]: LGTM, I'd recommend keeping your spacing consistent with nested options/data

File: db/seed.js [hunk 2]
```diff
@@ -3,14 +3,14 @@ const productDetails = require('./index.js').productDetails;
...
 
-productOptions = [
```
Comment (matt-leyrat) [inline db/seed.js]: Careful not to commit bugs :D

File: db/seed.js [hunk 3]
```diff
@@ -6,19 +6,19 @@ let productNames = ['Personalized Cheese Board Wedding Gift Set', 'Unpainted Dor
 let productOptions = [
   {
     optionName: 'Size',
-    choices: ['extra-small', 'small', 'medium', 'large', 'extra-large']
+    choices: [{choice: 'extra-small'}, {choice: 'small'}, {choice: 'medium'}, {choice: 'large'}, {choice:'extra-large'}]
   },
   {
     optionName: 'Material',
-    choices: ['ash', 'walnut', 'ebony', 'aluminum', 'brushed steel', 'glass', 'solid titanium', 'pure gold', 'solid diamond']
+    choices: [{choice: 'ash'}, {choice:'walnut'}, {choice:'ebony'}, {choice: 'aluminum'}, {choice: 'brushed steel'}, {choice: 'glass'}, {choice: 'solid titanium'}, {choice: 'pure gold'}, {choice: 'solid diamond'}]
```
Comment (matt-leyrat) [inline db/seed.js]: Why are we adding a choice key to each element? That'll probably turn into some awkward syntax (`productOptions.choices[0].choice`)

File: db/seed.js [hunk 4]
```diff
@@ -1,42 +1,22 @@
...
-    optionName: 'Pattern',
-    choices: [{choice: 'checkerboard'}, {choice: 'argile'}, {choice: 'striped-vertical'}, {choice: 'striped-horizontal'}, {choice: 'stars'}, {choice: 'bars'}, {choice: 'tie-dye'}]
-  },
-  {
-    optionName: 'Font Design',
-    choices: [{choice: 'serif'}, {choice: 'comic-sans'}, {choice: 'typewriter'}, {choice: 'cursive'}, {choice: 'star-wars'}]
-  }
-];
+//const productDetails = require('./index.js').productDetails;
+const dataSources = require('./fakedataSources.js');
```
Comment (matt-leyrat) [inline db/seed.js]: Good call moving it all to a different file 👍

File: db/seed.js [hunk 5]
```diff
@@ -95,5 +95,12 @@ class dataGenerator {
...
-productDetails.create(product);
+products = [];
+let product;
+for (let i = 0; i < 150; i++) {
+  product = generator.generateProduct();
+  products.push(product);
+}
+productDetails.create(products)
+  .then(() => mongoose.connection.close())
+  .catch((err) => console.log(err));
```
Comment (matt-leyrat) [inline db/seed.js]: Don't forget to add a new line at the end of every file ~ a linter tool would be helpful!

File: db/seed.js [hunk 6]
```diff
@@ -95,5 +95,12 @@ class dataGenerator {
 }
 
 let generator = new dataGenerator();
-let product = generator.generateProduct();
-productDetails.create(product);
+products = [];
+let product;
```
Comment (matt-leyrat) [inline db/seed.js]: I'm curious why we're declaring an empty variable and re-assigning it with every iteration of the loop

File: db/seed.js [hunk 7]
```diff
@@ -35,7 +35,7 @@ class dataGenerator {
   }
 
   generateProductId() {
-    return this.getRandomInt(1, 10000);
+    return this.getRandomInt(1, 100000);
```
Comment (matt-leyrat) [inline db/seed.js]: You might want to keep your product ID within the same range as your teammates (1-100 for example). I'd recommend looking into generating auto-incremented ID's when inserting into your database

File: db/index.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+const mongoose = require('mongoose');
+
+mongoose.connect('mongodb://localhost/BTetsy' ,{useNewUrlParser: true, useUnifiedTopology: true});
```
[the remark this answers is not in the document]
Comment (mflinn92) [reply db/index.js]: The version of mongoose I have installed was throwing deprecation warnings that said to use these options instead. It did work without them but I was sick of the warnings popping up so I added them

File: db/seed.js (hunk 7 above)
Comment (mflinn92) [reply db/seed.js]: Will do thanks!

**Pull request merged**